eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala

```diff
@@ -188,6 +189,7 @@ object NodeParams {
       maxHtlcValueInFlightMsat = UInt64(config.getLong("max-htlc-value-in-flight-msat")),
       maxAcceptedHtlcs = maxAcceptedHtlcs,
       expiryDeltaBlocks = config.getInt("expiry-delta-blocks"),
+      upstreamTimeoutSafetyBlocks = config.getInt("upstream-timeout-safety-blocks"),
```

Comment:
Should we check that `upstreamTimeoutSafetyBlocks` is strictly greater than `expiryDeltaBlocks`? If the cltv delta is smaller than the safety timeout the htlc will be considered "almost" timed out as soon as we receive a fulfill from downstream.

Reply:
That's the other way around :)
`upstreamTimeoutSafetyBlocks` needs to be strictly smaller than `expiryDeltaBlocks`, should I assert that in the code itself and prevent the node from starting if that's not satisfied?

Reply:
Oh indeed it has to be strictly smaller :), i think an assertion in `NodeParams` should be enough. The user configuring this parameter must know that if they are too close it risks to have unnecessary channel closures.